src/java/org/apache/cassandra/cql3/statements/DescribeStatement.java

```diff
@@ -133,11 +133,13 @@ public final ResultMessage execute(QueryState state, QueryOptions options, long
     @Override
     public ResultMessage executeLocally(QueryState state, QueryOptions options)
     {
-        Keyspaces keyspaces = Schema.instance.snapshot();
-        UUID schemaVersion = Schema.instance.getVersion();
+        Schema schema = SchemaManager.instance.schema();
+        Keyspaces keyspaces = schema.getKeyspaces();
+        UUID schemaVersion = schema.getVersion();
 
         keyspaces = Keyspaces.builder()
                              .add(keyspaces)
+                             .add(SchemaManager.instance.getLocalKeyspaces())
```

Comment:
ouch! Although I like the local/shard separation, usages where we need all of them may be prone to errors.